integrations/gcp/gcp_core/search/searches.py

```diff
@@ -0,0 +1,186 @@
+    The listing is being done via the PublisherAsyncClient, ignoring state in assets inventory
+    """
+    project_name = project["name"]
+    logger.info(
+        f"Searching all {AssetTypesWithSpecialHandling.TOPIC}'s in project {project_name}"
+    )
+    async with PublisherAsyncClient() as async_publisher_client:
+        try:
+            list_topics_pagers = await async_publisher_client.list_topics(
+                project=project_name
+            )
+            async for paginated_response in list_topics_pagers.pages:
+                topics = parse_protobuf_messages(paginated_response.topics)
+                if topics:
+                    logger.info(
+                        f"Generating {len(topics)} {AssetTypesWithSpecialHandling.TOPIC}'s"
+                    )
+                    for topic in topics:
+                        topic[EXTRA_PROJECT_FIELD] = project
+                    yield topics
+        except PermissionDenied:
+            logger.info(
+                f"Service account doesn't have permissions to list topics from project {project_name}"
+            )
+        except NotFound:
+            logger.debug(f"Project {project_name} is deleted")
+
+
+@cache_iterator_result()
+async def search_all_projects() -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """
+    Search for projects that the caller has ``resourcemanager.projects.get`` permission on
+    """
+    async with ProjectsAsyncClient() as projects_client:
+        search_projects_pager = await projects_client.search_projects()
+        async for projects_page in search_projects_pager.pages:
+            yield parse_protobuf_messages(projects_page.projects)
+
+
+async def search_all_folders() -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """
+    Search for folders that the caller has ``resourcemanager.folders.get`` permission on
+    """
+    async with FoldersAsyncClient() as folders_client:
+        search_folders_pager = await folders_client.search_folders()
+        async for folders_page in search_folders_pager.pages:
+            yield parse_protobuf_messages(folders_page.folders)
+
+
+async def search_all_organizations() -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """
+    Search for organizations that the caller has ``resourcemanager.organizations.get``` permission on
+    """
+    async with OrganizationsAsyncClient() as organizations_client:
+        search_organizations_pager = await organizations_client.search_organizations()
+        async for organizations_page in search_organizations_pager.pages:
+            yield parse_protobuf_messages(organizations_page.organizations)
+
+
+async def get_project(project_name: str) -> RAW_ITEM:
+    async with ProjectsAsyncClient() as projects_client:
+        return parse_protobuf_message(
+            await projects_client.get_project(name=project_name)
+        )
+
+
+async def get_folder(folder_name: str) -> RAW_ITEM:
```

Comment:
for every get single item can you please add something to the name that will say its single
maybe `get_single_{type}`